### ISSUE TITLE
yscv-onnx: YOLO model inference produces incorrect output

### ISSUE THREAD

**Issue by kakserpom** [enthropy7/YSCV#2]
## Bug Description

The yscv-onnx runtime produces incorrect output when running YOLO ONNX models:

### Issue 1: YOLOv8 - Scores outside [0,1] range

When running YOLOv8 models, the class confidence scores are raw logits (600+) instead of normalized probabilities [0,1]. This indicates the sigmoid activation is not being applied.

**Reproduction:**
```rust
let model = load_onnx_model_from_file("yolov8n.onnx")?;
let outputs = run_onnx_model(&model, inputs)?;
```

**Expected:** Class scores in range [0, 1]
**Actual:** Class scores around 600-650

### Issue 2: YOLO11 - Shape mismatch error

YOLO11 ONNX models fail with a shape decode error.

**Error:**
```
DecodeFailed { message: "shape mismatch: left=[1, 1], right=[1]" }
```

## Environment

- yscv version: latest (cloned 2026-03-23)
- Rust version: 1.88+
- Platform: macOS ARM64

## Models Tested

- `yolov8n_opset11.onnx` - YOLOv8 opset 11 export
- `yolo11n.onnx` - YOLO11 from Ultralytics

## Test Code

```rust
#[test]
fn test_yscv_onnx_output_analysis() {
    let model = load_onnx_model_from_file("yolov8n_opset11.onnx").unwrap();
    let outputs = run_onnx_model(&model, inputs).unwrap();
    
    // Output shape: [1, 84, 8400] - correct
    // But scores are ~600-650 instead of 0-1
}
```

## Suggested Fix

1. Ensure sigmoid activation is properly applied after model inference
2. Fix shape handling for YOLO11 output tensors
3. Or document that external ONNX runtime (ort crate) should be used for YOLO models

## Workaround

Use the `ort` crate for YOLO model inference instead of yscv-onnx.

**Comment by enthropy7** [comment]
scores are correct - sigmoid IS applied in the graph. the output [1, 84, 8400] is in YOLOv8 format (features × predictions), not [1, 8400, 84]:

Rows 0–3: cx, cy, w, h — bbox pixel coordinates (0–640)
Rows 4–83: class confidence scores — already sigmoid'd, range [0, 1]
Your score=631.99 is a bbox coordinate being read as a score. The output shape [1, 84, 8400] means 84 < 8400, so it's YOLOv8 layout regardless of the model name.

**Correct decode:**

// For prediction i:
```
let cx = data[0 * 8400 + i];       // row 0
let cy = data[1 * 8400 + i];       // row 1
let w  = data[2 * 8400 + i];       // row 2
let h  = data[3 * 8400 + i];       // row 3
let score = data[(4 + class) * 8400 + i];  // rows 4-83, already [0,1]
```

**Or use the built-in auto-detect:**
```

if shape[1] < shape[2] {
    decode_yolov8_output(output, &config, w, h)  // [1, 84, 8400]
} else {
    decode_yolov11_output(output, &config, w, h) // [1, 8400, 84]
}
```
